src/generate/convert/control_flow.rs

```diff
@@ -7,26 +7,49 @@ use crate::generate::result::{GenResult, UnimplementedErr};
 
 pub fn convert_cntrl_flow(ast: &ASTTy, imp: &mut Imports, state: &State, ctx: &Context) -> GenResult {
     Ok(match &ast.node {
-        NodeTy::IfElse { cond, then, el } => match el {
-            Some(el) => Core::IfElse {
-                cond: Box::from(convert_node(cond, imp, &state.last_ret(false), ctx)?),
-                then: Box::from(convert_node(then, imp, state, ctx)?),
-                el: Box::from(convert_node(el, imp, state, ctx)?),
-            },
-            None => Core::If {
-                cond: Box::from(convert_node(cond, imp, state, ctx)?),
-                then: Box::from(convert_node(then, imp, state, ctx)?),
-            },
-        },
+        NodeTy::IfElse { cond, then, el } => {
+            let cond = Box::from(convert_node(cond, imp, &state.is_last_must_be_ret(false).must_assign_to(None), ctx)?);
```

Comment:
A bit sloppy to have to reset state like this constantly, but I can't think of another way to do this.